### PR TITLE
Update several notebooks to use cleaner `odc-geo` code

### PR DESCRIPTION
**Pull request by robbibt**
### Proposed changes
The [Open Data Cube Geo / `odc-geo` package](https://odc-geo.readthedocs.io/en/latest/intro.html) now includes functionality to make it easy to perform spatial processing tasks using its [built-in `.odc.` methods](https://odc-geo.readthedocs.io/en/latest/api.html#odc-geo-xr) that are automatically added to `xarray` data. 

For example, we can now mask an `xarray` dataset by a polygon using:
```
import odc.geo.xr
ds.odc.mask(poly)
```
Or reprojecting a dataset:
```
ds.odc.reproject(how="EPSG:3577")
```

This PR updates several notebooks to take advantage of this new functionality and simplify our code.

It also adds a new `load_reproject` function which uses `odc.geo` and `rasterio` to load and reproject external rasters to match data loaded from DEA.

### Closes issues (optional)
- Closes Issue #000

### Checklist 
(Replace `[ ]` with `[x]` to check off)

- [x] Notebook created using the [DEA-notebooks template](https://github.com/GeoscienceAustralia/dea-notebooks/tree/develop)
- [x] Remove any unused Python packages from `Load packages`
- [x] Remove any unused/empty code cells
- [x] Remove any guidance cells (e.g. `General advice`)
- [x] Ensure that all code cells follow the [PEP8 standard](https://www.python.org/dev/peps/pep-0008/) for code. The `jupyterlab_code_formatter` tool can be used to format code cells to a consistent style: select each code cell, then click `Edit` and then one of the `Apply X Formatter` options (`YAPF` or `Black` are recommended).
- [x] Include relevant tags in the final notebook cell (refer to the [DEA Tags Index](https://docs.dea.ga.gov.au/genindex/), and re-use tags if possible)
- [x] Clear all outputs, run notebook from start to finish, and save the notebook in the state where all cells have been sequentially evaluated
- [ ] Test notebook on both the `NCI` and `DEA Sandbox` (flag if not working as part of PR and ask for help to solve if needed)
- [ ] If applicable, update the `Notebook currently compatible with the NCI|DEA Sandbox environment only` line below the notebook title to reflect the environments the notebook is compatible with
- [x] Check for any spelling mistakes using the DEA Sandbox's built-in spellchecker (double click on markdown cells then right-click on pink highlighted words). For example:

sandbox_spellchecker

### CODE REVIEW

File: How_to_guides/Analyse_multiple_polygons.ipynb [hunk 1]
```diff
@@ -4,7 +4,7 @@
    "cell_type": "markdown",
```
Comment (BexDunn) [inline How_to_guides/Analyse_multiple_polygons.ipynb:4]: <p>Hi, these comments seem to be from the old notebook : &nbsp;&nbsp;</p><p> </p><p># Append results to a dictionary using the attribute</p><p>&nbsp;&nbsp;# column as an key</p><p>&nbsp;&nbsp;# results.update({str(row[attribute_col]): ds_masked})</p><p> </p><p>Suggest updating to reflect new usage?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/1186/discussion/#comment-1495095897'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (BexDunn) [inline How_to_guides/Analyse_multiple_polygons.ipynb:4]: <p>We're getting a runtime value here, and a quick search suggests that matplotlib is flagging us trying to cast a NaN to a colour in the colourmapping function (and not handling the error). Do you think that's correct? Is this something we should flag with matplotlib or some difference between odc.geo and datacube.utils in specifying number formats? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/1186/discussion/#comment-1495095900'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (BexDunn) [reply How_to_guides/Analyse_multiple_polygons.ipynb:4]: image

**Review by BexDunn — APPROVED** [review]
Looks great, and very useful. Two comments on the Analyse multiple polys notebook.
First, a couple of comments in the text to update: 
# Append results to a dictionary using the attribute
  # column as an key
  # results.update({str(row[attribute_col]): ds_masked})

And secondly, something funky happening with the error message on the plot related to casting numbers? - more details in the comments, and reviewNB

**Review by BexDunn — CHANGES_REQUESTED** [review]
argh, please see other review! I meant to request changes :)

**Comment by robbibt** [comment]
Hey @BexDunn, I fixed up the comment - but when I re-ran the notebook, the runtime warning disappeared. I've had that happen before - it's a confusing! Can keep an eye out for it if it re-appears

**Review by BexDunn — APPROVED** [review]

**Pull request merged**